apps/emqx_durable_storage/src/emqx_ds_message_storage_bitmask.erl

```diff
@@ -340,22 +341,30 @@ next(It0 = #it{filter = #filter{keymapper = Keymapper}}) ->
             {error, closed}
     end.
 
--spec preserve_iterator(iterator()) -> binary().
-preserve_iterator(#it{cursor = Cursor}) ->
+-spec preserve_iterator(iterator()) -> serialized_iterator().
+preserve_iterator(#it{
+    cursor = Cursor,
+    filter = #filter{
+        topic_filter = TopicFilter,
+        start_time = StartTime
+    }
+}) ->
     State = #{
         v => 1,
-        cursor => Cursor
+        cursor => Cursor,
+        replay => {TopicFilter, StartTime}
```

Comment:
Sorry, just passing by, decided to share why I wanted to avoid that.
1. This part of a serialized iterator never changes.
2. As an original topic filter, it would probably be the biggest part of the serialized binary.
3. This information is most likely already stored elsewhere (i.e. as part of session record).

Assuming that each iterator needs to be written to the storage on each (or almost each) ACK, points 1 and 2 combined would probably lead to a (comparatively) massive waste of write bandwidth and storage capacity.

Reply:
Good points.

However, these changes might already be out of date by now.  :sweat_smile: 

This commit was incorporated by #11720 , and @ieQu1 changed a lot the iterator API, so this might not be needed anymore after that refactoring is done.